src/main/java/org/jabref/preferences/FilePreferences.java

```diff
@@ -45,7 +43,7 @@ public String getUser() { // Read only
     }
 
     public Optional<Path> getFileDirectory() {
-        if (StringUtil.isBlank(mainFileDirectory.getValue())) {
+        if (storeFilesRelativeToBibFile.getValue()) {
             return Optional.empty();
         } else {
             return Optional.of(Path.of(mainFileDirectory.getValue()));
```

Comment:
This maybe fix the bug, the Preferences Object is no simple dataholder anymore and violates the single responsibility principle imho. The get method should just return the value, to check if storeFilesRelativeToBibFile is checked must be checked in the logic.

Reply:
There's a check in logic for optional empty and also for should store relative
Need to check if I can adjust the logic.